applications/FSIApplication/custom_utilities/ibqn_mvqn_convergence_accelerator.h

```diff
@@ -0,0 +1,455 @@
+            {
+                TDenseSpace::GetColumn(Col, *p_inv_jac_left, rAuxColumnTLS);
+                for (std::size_t row = 0; row < problem_size; ++row) {
+                    aux_LHS(row,Col) -= TDenseSpace::RowDot(row, *p_inv_jac_right, rAuxColumnTLS);
+                }
+            });
+
+            // Calculate the correction
+            // Do the QR decomposition of (I - J_{S}J_{F}) and solve for the force update
+            QR<double, row_major> qr_util;
+            qr_util.compute(problem_size, problem_size, &(aux_LHS)(0,0));
+            qr_util.solve(&(aux_RHS)(0), &(right_correction)(0));
+        }
+
+        // Update the iteration guess
+        TSparseSpace::UnaliasedAdd(rIterationGuess, 1.0, right_correction);
+    }
+
+    //TODO: CREATE AN INTERMEDIATE IBQN BASE CLASS
+    virtual void UpdateSolutionLeft(
+        const VectorType& rDisplacementInputVector,
+        const VectorType& rForceOutputVector,
+        VectorType& rIterationGuess)
+    {
+        KRATOS_TRY;
+
+        // Save the uncorrected force vector for the displacement update
+        auto p_uncor_force_aux = Kratos::make_shared<VectorType>(rForceOutputVector);
+        std::swap(mpUncorrectedForceVector, p_uncor_force_aux);
+
+        // Update left inverse Jacobian approximation with the current information
+        mpConvergenceAcceleratorLeft->UpdateInverseJacobianApproximation(rDisplacementInputVector, rForceOutputVector);
+
+        VectorType left_correction(TSparseSpace::Size(rIterationGuess));
+        if (!mFirstLeftCorrectionPerformed) {
+            // Do nothing in the first traction correction
+            // This means to set as correction the difference between the guess and the previous iteration
+            left_correction = rForceOutputVector - rIterationGuess;
+
+            // Update the first iteration correction flag
+            mFirstLeftCorrectionPerformed = true;
+
+        } else {
+            // Get the interface problem size
+            std::size_t problem_size = mpConvergenceAcceleratorLeft->GetProblemSize();
+
+            // Get both left and inverse Jacobian
+            auto p_inv_jac_left = mpConvergenceAcceleratorLeft->pGetInverseJacobianApproximation();
+            auto p_inv_jac_right = mpConvergenceAcceleratorRight->pGetInverseJacobianApproximation();
+
+            // Set the residual of the update problem to be solved
+            VectorType aux_RHS = rForceOutputVector - rIterationGuess;
+            VectorType aux_left_onto_right(problem_size);
+            VectorType displacement_iteration_update = *mpUncorrectedDisplacementVector - rDisplacementInputVector;
+            TSparseSpace::Mult(*p_inv_jac_left, displacement_iteration_update, aux_left_onto_right);
+            TSparseSpace::UnaliasedAdd(aux_RHS, 1.0, aux_left_onto_right);
+
+            // Set the LHS of the update problem to be solved
+            MatrixType aux_LHS = IdentityMatrix(problem_size, problem_size);
```

Comment:
ok I guess there is really no way to save this guy in terms of memory :D

Maybe makes sense to store this as a member to avoid constant reallocations?

Reply:
:smile: 

I've discussing again with @RiccardoRossi . This is a very first working but not optimal implementation,  we wonder about the possibility of implementing the projection of the Jacobian onto a vector rather than exposing it.

In any case we think that we should first merge this (together with the changes in the solvers to enable using it) to have a framework to put the hands on. The nice thing would be that all the future improvements will be located at the "algebra" level rather than in the usage.